core/listener.go

```diff
@@ -62,75 +67,106 @@ func (cl *Listener) Start(context.Context) error {
 	if err != nil {
 		return err
 	}
-	go cl.listen(ctx, sub)
+	go cl.runSubscriber(ctx, sub)
 	return nil
 }
 
 // Stop stops the listener loop.
-func (cl *Listener) Stop(ctx context.Context) error {
+func (cl *Listener) Stop(context.Context) error {
 	cl.cancel()
 	cl.cancel = nil
-	return cl.fetcher.UnsubscribeNewBlockEvent(ctx)
+	return nil
+}
+
+// runSubscriber runs a subscriber to receive event data of new signed blocks. It will attempt to
+// resubscribe in case error happens during listening of subscription
+func (cl *Listener) runSubscriber(ctx context.Context, sub <-chan types.EventDataSignedBlock) {
+	for {
+		err := cl.listen(ctx, sub)
```

Comment:
Seems like we are not doing anything here with actual error.

Reply:
We log warn it and resubscribe right after. I'